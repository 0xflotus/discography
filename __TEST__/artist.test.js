import testSnapshot from './fixtures/testSnapshot';

test('Shoulds have all basic fields on the artist type', async () => {
  const query = `
    query {
      lookup {
        artist (id: 1) {
          id
          name
          realName
          aliases
          urls
        }
      }
    }
  `;

  await testSnapshot(query);
});

test('Should implement the artist-release connection', async () => {
  const query = `
    query {
      lookup {
        artist (id: 1) {
          id
          releases (first: 1) {
            edges {
              node {
                id
                title
              }
            }
          }
        }
      }
    }
  `;

  await testSnapshot(query);
});
