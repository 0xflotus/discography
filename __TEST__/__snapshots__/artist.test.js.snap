// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should implement the artist-release connection 1`] = `
Object {
  "data": Object {
    "lookup": Object {
      "artist": Object {
        "id": 1,
        "releases": Object {
          "edges": Array [
            Object {
              "node": Object {
                "id": 1,
                "title": "Stockholm",
              },
            },
          ],
        },
      },
    },
  },
}
`;

exports[`Shoulds have all basic fields on the artist type 1`] = `
Object {
  "data": Object {
    "lookup": Object {
      "artist": Object {
        "aliases": Array [
          "Dick Track",
          "Faxid",
          "Groove Machine",
          "Janne Me' Amazonen",
          "Jesper Dahlbäck",
          "Lenk",
          "The Pinguin Man",
        ],
        "id": 1,
        "name": "The Persuader",
        "realName": "Jesper Dahlbäck",
        "urls": null,
      },
    },
  },
}
`;
